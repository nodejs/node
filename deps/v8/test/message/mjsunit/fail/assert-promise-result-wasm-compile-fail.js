// Copyright 2019 the V8 project authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

// Flags: --experimental-stack-trace-frames

// Test wasm compilation explicitly, since this creates a promise which is only
// resolved later, i.e. the message queue gets empty in-between.
// The important part here is that d8 exits with a non-zero exit code.

d8.file.execute('test/mjsunit/mjsunit.js');
d8.file.execute('test/mjsunit/wasm/wasm-module-builder.js');

assertPromiseResult((async function test() {
  let ok_buffer = (() => {
    let builder = new WasmModuleBuilder();
    builder.addFunction(undefined, kSig_i_v).addBody([kExprI32Const, 42]);
    return builder.toBuffer();
  })();
  let bad_buffer = new ArrayBuffer(0);
  let kNumCompiles = 3;

  // Three compilations of the OK module should succeed.
  for (var i = 0; i < kNumCompiles; ++i) {
    await WebAssembly.compile(ok_buffer);
  }

  // Three compilations of the bad module should fail.
  for (var i = 0; i < kNumCompiles; ++i) {
    await WebAssembly.compile(bad_buffer);
  }
})());
