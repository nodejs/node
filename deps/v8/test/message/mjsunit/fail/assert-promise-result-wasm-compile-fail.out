test/mjsunit/mjsunit.js:{NUMBER}: CompileError: WebAssembly.compile(): BufferSource argument is empty
              throw concatenateErrors(stack, e);
              ^
Error
    at assertPromiseResult (test/mjsunit/mjsunit.js:{NUMBER}:{NUMBER})
    at *%(basename)s:{NUMBER}:1

CompileError: WebAssembly.compile(): BufferSource argument is empty
    at WebAssembly.compile (<anonymous>)
    at test (*%(basename)s:{NUMBER}:23)

